src/components/clusterConfiguration/ClusterConfiguration.tsx

```diff
@@ -154,6 +155,7 @@ const ClusterConfiguration: React.FC<ClusterConfigurationProps> = ({ cluster })
     ingressVip: cluster.ingressVip || '',
     pullSecret: '',
     sshPublicKey: cluster.sshPublicKey || '',
+    pullSecretSet: !!cluster.pullSecretSet, // private (read-only), for validations only, set by backend
```

Comment:
Since we're introducing this, we don't need to keep the value in the component in the state (isPullSecretEdit, setPullSecretEdit) any more. We can replace those with `values.pullSecretSet` and `setFieldValue('pullSecretSet',...)

Reply:
done